CDCgov/prime-reportstream
# 7047 - Added functionality to support the reverseQualityFilter flag

Added functionality to support the reverseQualityFilter flag for receivers

Test Steps:
1. Submit a Full ELR HL7 using reports API - /api/reports?processing=async&default=aaa
2. Capture the Id of the submitted report
3. Verify Full ELR HL7 destination using report history API - /waters/report/{{reportId}}/history
4. Capture the receiver being routed
5. Update the reverseTheQualityFilter flag on this receiver inside organizations.yaml
6. Run ./gradlew reloadSettings to update settings for this receiver
7. Repeat steps 1 and 2
8. Verify the destination the previous destination is not present when calling the report history API

## Changes
- Added optional parameter reverseFilter to the evaluateFilterCondition
- Passing reverseTheQualityFilter flag when evaluating quality filters

## Checklist

### Testing

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #7047
